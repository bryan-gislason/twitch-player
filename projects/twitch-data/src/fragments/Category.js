import gql from "graphql-tag";
import TagFragment from "./Tag";

export default gql`
  fragment Category on Game {
    id
    key: id
    name
    boxArtURL
    viewersCount
    followersCount
  }
  ${TagFragment}
`;
