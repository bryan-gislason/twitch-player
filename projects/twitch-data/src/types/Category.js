import PropTypes from "prop-types";
import TagType from "./Tag";

export default PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  boxArtURL: PropTypes.string.isRequired,
  viewersCount: PropTypes.number.isRequired,
  followersCount: PropTypes.number.isRequired,
  tags: PropTypes.arrayOf(TagType),
});
