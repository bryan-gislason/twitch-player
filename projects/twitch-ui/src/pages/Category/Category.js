import PropTypes from "prop-types";
import React, { memo } from "react";
import CategoryChannels from "../../components/CategoryChannels";

const Category = ({ match }) => {
  const { name } = match.params;

  return <CategoryChannels name={name} heading={name} />;
};

Category.propTypes = {
  match: PropTypes.object.isRequired,
};

export default memo(Category);
