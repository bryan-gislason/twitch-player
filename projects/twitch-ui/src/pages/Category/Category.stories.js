import { storiesOf } from "@storybook/react";
import { createClient } from "@twitch-player/data";
import React from "react";
import { ApolloProvider } from "react-apollo-hooks";
import Main from "../../components/Main";
import Category from ".";

storiesOf("Pages|Category", module).add("default", () => (
  <ApolloProvider client={createClient()}>
    <Main>
      <Category match={{ params: { name: "Fortnite" } }} />
    </Main>
  </ApolloProvider>
));
