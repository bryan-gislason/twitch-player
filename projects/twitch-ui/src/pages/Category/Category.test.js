import React from "react";
import { render } from "../../testing";
import Category from "./Category";

jest.mock("../../components/CategoryChannels", () => "mock-category-channels");

describe("Category", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Category match={{ params: { name: "Fortnite" } }} />);
  });

  it("renders the page", () => {
    expect(instance.asFragment()).toMatchSnapshot();
  });
});
