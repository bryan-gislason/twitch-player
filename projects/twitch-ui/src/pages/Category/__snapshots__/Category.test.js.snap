// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders the page 1`] = `
<DocumentFragment>
  <mock-category-channels
    heading="Fortnite"
    name="Fortnite"
  />
</DocumentFragment>
`;
